compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -789,13 +789,15 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
 
                 Some(args.constraints.iter().filter_map(|constraint| {
                     let ident = constraint.ident;
-                    let trait_def = path.res.def_id();
-                    let assoc_item = tcx.associated_items(trait_def).find_by_name_and_kind(
-                        tcx,
-                        ident,
-                        ty::AssocKind::Type,
-                        trait_def,
-                    );
+                    let trait_def = path.res.opt_def_id();
+                    let assoc_item = trait_def.and_then(|trait_def_id| {
```

Comment:
Please leave a comment here saying that we don't want to unwrap a trait def id if it resolves incorrectly.